tensorflow-utils/src/main/java/org/tensorflow/util/buffer/FloatDataBuffer.java

```diff
@@ -0,0 +1,153 @@
+/*
+ Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ =======================================================================
+ */
+package org.tensorflow.util.buffer;
+
+import java.nio.BufferOverflowException;
+import java.nio.BufferUnderflowException;
+import java.nio.ReadOnlyBufferException;
+
+/**
+ * A {@link DataBuffer} of floats.
+ */
+public interface FloatDataBuffer extends DataBuffer<Float> {
+
+  /**
+   * Reads the float at the given index.
+   *
+   * @param index the index from which the float will be read
+   * @return the float at the given index
+   * @throws IndexOutOfBoundsException if index is negative or not smaller than the buffer size
+   */
+  float getFloat(long index);
```

Comment:
I like these methods, as it allows us to make an FP16 one that can have an FP16 view (either as as short or a value type) and a float view. So I don't think they should be renamed to just `get/set`.